pkg/cli/preflight.go

```diff
@@ -0,0 +1,154 @@
+package cli
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kurl/pkg/installer"
+	"github.com/replicatedhq/kurl/pkg/preflight"
+	analyze "github.com/replicatedhq/troubleshoot/pkg/analyze"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func NewPreflightCmd(cli CLI) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "preflight [installer spec file]",
+		Short:        "Runs kURL preflight checks",
+		SilenceUsage: true,
+		Args:         cobra.ExactArgs(1),
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			return viper.BindPFlags(cmd.PersistentFlags())
+		},
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return viper.BindPFlags(cmd.Flags())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			installerSpec, err := installer.RetrieveSpec(cli.GetFS(), args[0])
+			if err != nil {
+				return errors.Wrap(err, "retrieve installer spec")
+			}
+
+			builtin := preflight.Builtin()
+			data := installer.TemplateData{
+				Installer: *installerSpec,
+				IsPrimary: viper.GetBool("is-primary"),
```

Comment:
I dont love all these flags but i cant think of a better way to do it. Open to suggestions.